### DISCUSSION TITLE
First one, simple text / string transformations

### DISCUSSION THREAD

**Discussion by letanure** [johnlindquist/kit#227] [Share]
Just created a simple one, no deps, for simple text transformations

- Remove duplicated spaces
- Uppercase [AAA BBB]
- Lowercase [aaa bbb]
- Capitalize words [Aaa Bbb]
- Snake case [aaa_bbb]
- kebab-case [aaa-bbb]
- camelCase [aaaBbb]
- PascalCase [AaaBbb]
- Remove numbers [aa1aBb2b => aaaBbb]
- Keep only numbers [aa1aBb2b => 12]
- Remove HTML tags
- Remove HTML attributes
- Remove accents & diacritics [âêì => aei]
- URL slugfy

https://codesandbox.io/s/funny-sun-zt1ur?file=/src/index.js

https://gist.github.com/letanure/ee7db17b3b381966f61aab9812862870

https://user-images.githubusercontent.com/81244/116789299-7cd07600-aaae-11eb-9fa2-3bbfa58ff053.mp4

**Comment by johnlindquist** [comment]
Nice work!

This may be a fun script to try out "Shortcodes":

https://github.com/johnlindquist/kit/discussions/194

**Reply by letanure** [reply]
like this?
https://gist.github.com/letanure/4378f3ac66653a23d8620468ec3d62de

**Reply by johnlindquist** [reply]
Yeah, that's definitely the idea. If this is something someone's going to do frequently, Shortcodes should help them commit it to muscle memory.

**Comment by crissxross** [comment]
I can't get this script to work. Here's the error log:

text not found
Error: text not found
    at resolveToScriptPath (file:///Users/christine/.kit/core/utils.js:144:11)
    at run (file:///Users/christine/.kit/core/utils.js:384:26)
    at kit (file:///Users/christine/.kit/index.js:11:25)
    at async file:///Users/christine/.kenv/scripts/text-transformations.js

**Reply by johnlindquist** [reply]
@crissxross remove line 6:

```js
let { getSelectedText, setSelectedText } = await kit("text")
```

If you find an "older" script with `await kit` anywhere, you can just delete that line. All those utils were made global.
